### PR TITLE
Docs: clarify alleles list with missing data

### PR DESCRIPTION
**Pull request by hyanwong**

### CODE REVIEW

File: tsinfer/formats.py [hunk 1]
```diff
@@ -1832,8 +1832,11 @@ def add_site(
             that the input array is also of this type.
         :param list(str) alleles: A list of strings defining the alleles at this
             site. Only biallelic sites can currently be used for inference. Sites
-            with 3 or more alleles cannot have ``inference`` (below) set to ``True``.
-            If not specified or None, defaults to ["0", "1"].
+            with 3 or more non-missing alleles cannot have ``inference`` (below)
+            set to ``True``. If missing data is present in the ``genotypes`` array,
```
Comment (jeromekelleher) [inline tsinfer/formats.py:1836]: This is just confusing it more by putting it in the params list, isn't it? Don't we just modify the returned list on the fly to make sure that -1 gives None, in the ``variants()`` iterator?

Comment (hyanwong) [reply tsinfer/formats.py:1836]: We add it to the list when we call add_site. That means we can use the arrays correctly too:

https://github.com/tskit-dev/tsinfer/blob/816593021c224bedc56c60744f6a3201d6f5f8cc/tsinfer/formats.py#L1889

Comment (jeromekelleher) [reply tsinfer/formats.py:1836]: Ah, ok

**Review by jeromekelleher — APPROVED** [review]

**Review by benjeffery — APPROVED** [review]

**Comment by benjeffery** [comment]
@hyanwong I've pushed a lint fix here.

**Comment by benjeffery** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**